Makefile

```diff
@@ -14,20 +14,63 @@
 # limitations under the License.
 #
 
-#all: test
+# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
+CRD_OPTIONS ?= "crd:trivialVersions=true,preserveUnknownFields=false"
 
-.PHONY: build
-build:
-	CGO_ENABLED=1 GOOS=linux go build     -o build/_output/bin/ibm-block-csi-operator     -gcflags all=-trimpath=${GOPATH} -asmflags all=-trimpath=${GOPATH} -mod=vendor cmd/manager/main.go
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+SHELL = /usr/bin/env bash -o pipefail
+.SHELLFLAGS = -ec
+
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+##@ Development
+
+manifests: controller-gen kustomize## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
+	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=ibm-block-csi-operator webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+	$(KUSTOMIZE) build config/crd > config/crd/bases/csi.ibm.com_ibmblockcsis_tmp.yaml
+	mv config/crd/bases/csi.ibm.com_ibmblockcsis_tmp.yaml config/crd/bases/csi.ibm.com_ibmblockcsis.yaml
```

Comment:
why do we need the tmp file?

Reply:
and what if we add more files in `config/crd`? shouldn't this be some `for` loop?

Reply:
> why do we need the tmp file?

because when I tried to overwrite the original file it left empty file

Reply:
oh, I guess it's because the build command reads from the non-temp file after the redirect is executed (truncating the file).
try using [`-o`](https://github.com/kubernetes-sigs/kustomize/blob/7b5e43d343ebf2fdb29c7f6a6af072c895b26d2f/kustomize/commands/build/flagoutputpath.go#L14) instead

Reply:
remember to address the [second question](https://github.com/IBM/ibm-block-csi-operator/pull/195#discussion_r689054682) as well

Reply:
> I didn't find a way to update a each resource if you have multiple resources

have you tried [specifying the **directory** for -o](https://github.com/kubernetes-sigs/kustomize/issues/3175#issuecomment-721429322)?

Reply:
it creates two files but it's not updating the crd yaml file, it creates another one `apiextensions.k8s.io_v1_customresourcedefinition_ibmblockcsis.csi.ibm.com.yaml`

Reply:
you're [right](https://github.com/kubernetes-sigs/kustomize/issues/2393#issue-604335999).
could we change the input filename to be the same as the output filename?

Reply:
I didn't find a way to do it :confused:

Reply:
:( then you could open a ticket for it?

Reply:
CSI-3315